### ISSUE TITLE
store a value in session

### ISSUE THREAD

**Issue by usamamashkoor** [keystonejs/keystone#4844]
Hi i want to store a access token from another api to the keystone session i have looked into the api docs but could not find anything related to this can someone let me know how can i do this in keystone.js

I have tried this
`req.session.accessToken1 = "mytoken";`

but when i go to another page it does not show my value mean it is not storing the value in the session.

kindly help me on this.

Thanks in advance.

**Comment by dyllandry** [comment]
Keystone implements [Express](https://expressjs.com/) for routing, and Express does not provide session handling features out of the box. Express does, however, support [middleware](https://expressjs.com/en/guide/using-middleware.html) packages that do, for example [express-session](https://www.npmjs.com/package/express-session). I am not sure how much of Express Keystone exposes, but after following the [Keystone Setting Up From Scratch](https://keystonejs.com/getting-started/setting-up/part-1) tutorial I see two places Express surfaces. In routes/index.js it appears the Express `app` is available as a parameter, there you could maybe utilize `app.use(...)` to attach the session middleware. And second, in routes/views/index.js where you can access the `req` and `res` objects, where you would access the session. 

In summary, I'm not sure if you can, though I have faith that the Keystone developers thought of that. Given the size of some of the companies that use Keystone, you probably can access/store session data. I hope I provided you some starting points, good luck!

**Comment by max8hine** [comment]
@usamamashkoor 
`req.session.accessToken1 = "mytoken"` should just work fine.

**Comment by autoboxer** [comment]
@usamamashkoor, were you able to figure out a solution?
